src/main/scala/com/mwz/sonar/scala/scapegoat/ScapegoatSensor.scala

```diff
@@ -0,0 +1,177 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala
+package scapegoat
+
+import inspections.ScapegoatInspection.AllScapegoatInspections
+
+import cats.implicits._
+import com.mwz.sonar.scala.util.JavaOptionals._
+import com.mwz.sonar.scala.util.Log
+import org.sonar.api.batch.fs.{FileSystem, InputFile}
+import org.sonar.api.batch.rule.ActiveRule
+import org.sonar.api.batch.sensor.{Sensor, SensorContext, SensorDescriptor}
+import org.sonar.api.batch.sensor.issue.NewIssue
+import org.sonar.api.config.Configuration
+import scalariform.ScalaVersion
+
+import java.nio.file.{Path, Paths}
+import scala.util.{Failure, Success, Try}
+
+/** Main sensor for importing Scapegoat reports to SonarQube */
+final class ScapegoatSensor(scapegoatReportParser: ScapegoatReportParserAPI) extends Sensor {
+  import ScapegoatSensorInternal._ // scalastyle:ignore scalastyle_ImportGroupingChecker
+
+  private[this] val log = Log(classOf[ScapegoatSensor], "scapegoat")
+
+  /** Populates the descriptor of this sensor */
+  override def describe(descriptor: SensorDescriptor): Unit =
+    descriptor
+      .createIssuesForRuleRepository(ScapegoatRulesRepository.RepositoryKey)
+      .name(SensorName)
+      .onlyOnFileType(InputFile.Type.MAIN)
+      .onlyOnLanguage(Scala.LanguageKey)
+
+  /** Saves the Scapegoat information of a module */
+  override def execute(context: SensorContext): Unit = {
+    log.info("Initializing the scapegoat sensor.")
+
+    val reportPath = getScapegoatReportPath(context.config)
+    log.info(s"Loading the scapegoat report file: '$reportPath'.")
+    Try(scapegoatReportParser.parse(reportPath)) match {
+      case Success(scapegoatIssuesByFilename) =>
+        log.info("Successfully loaded the scapegoat report file.")
+        processScapegoatWarnings(context, scapegoatIssuesByFilename)
+      case Failure(ex) =>
+        log.error(
+          "Aborting the scapegoat sensor execution, " +
+          s"cause: an error occurred while reading the scapegoat report file: '$reportPath', " +
+          s"the error was: ${ex.getMessage}."
+        )
+    }
+  }
+
+  /** Returns the path to the scapegoat report for this module */
+  private[scapegoat] def getScapegoatReportPath(settings: Configuration): Path = {
+    val scalaVersion = Scala.getScalaVersion(settings)
+    val defaultScapegoatReportPath = getDefaultScapegoatReportPath(scalaVersion)
+
+    if (!settings.hasKey(ScapegoatReportPathPropertyKey)) {
+      log.info(
+        s"Missing the property: '$ScapegoatReportPathPropertyKey', " +
+        s"using the default value: '$defaultScapegoatReportPath'."
+      )
+    }
+
+    settings
+      .get(ScapegoatReportPathPropertyKey)
+      .toOption
+      .map(path => Paths.get(path))
+      .getOrElse(defaultScapegoatReportPath)
+  }
+
+  /** Process all scapegoat warnings */
+  private[this] def processScapegoatWarnings(
```

Comment:
Ideally, we should also cover this by unit tests.

Reply:
Well, I was thinking about that...
But, since the current implementation of the sensor only calls that method, I don't see any value in having two separate tests that do the same. _(testing that method directly would make unnecessary to mock the report parser by the way)_.

Reply:
ok, that's fair enough